modules/engine/src/main/scala/observe/engine/Sequence.scala

```diff
@@ -10,25 +10,33 @@ import observe.engine.Action.ActionState
 import observe.engine.Result.RetVal
 import observe.model.SequenceState
 import lucuma.core.model.Observation
-import lucuma.core.model.sequence.Step.{Id => StepId}
+import lucuma.core.model.sequence.Atom as OcsAtom
+import lucuma.core.model.sequence.Step.Id as StepId
 
 /**
  * A list of `Step`s grouped by target and instrument.
  */
-final case class Sequence[F[_]](
-  id:    Observation.Id,
-  steps: List[Step[F]]
+final case class Sequence[F[_]] private (
+  id:     Observation.Id,
+  atomId: Option[OcsAtom.Id],
```

Comment:
The `Sequence` is a flattened list of `Steps`? Why is the `Atom` optional?

Reply:
To allow for the special case of an empty Sequence. For example, one that has run to completion.